minard/views.py

```diff
@@ -546,6 +547,27 @@ def l2():
         return redirect(url_for('l2',step=step,height=height,_external=True))
     return render_template('l2.html',step=step,height=height)
 
+@app.route('/burst')
+def burst():
+    return render_template('burst.html',data=load_burst_runs())
+
+def load_burst_runs():
+    server = couchdb.Server("https://snoplus:"+app.config["COUCHDB_PASSWORD"]+"@"+app.config["COUCHDB_HOSTNAME"])
+    db = server["burst"]
+
+    results = {}
+    i = 0 #the counter keeps the ordering correct
+    for row in db.view('_design/burst/_view/burst_by_run', descending=False):
+        run = row.key[0]
+        run_id = row.id
+        try:
+            results[i] = dict(db.get(run_id).items())
+        except KeyError:
+            app.logger.warning("Code returned KeyError searching for burst information in the couchDB. Run Number: %d" % run)
+        i += 1
```

Comment:
Shouldn't results just be a list? I'm not sure I understand the advantage of using a dictionary here with keys that are just an incrementing integer.

Reply:
Since the 'usual' logic uses IDs as key, these can get out of order being stored in dictionaries in this way. The results come ordered from the couch and I wanted to keep the ordering. Therefore I could either set the keys to be run_subrun_burst (as they are in the view) or just use the simple dummy iterator. This seemed simpler.